coreax/metrics.py

```diff
@@ -95,86 +91,104 @@ def __init__(self, kernel: coreax.kernel.Kernel, precision_threshold: float = 1e
 
     def compute(
         self,
-        x: ArrayLike,
-        y: ArrayLike,
+        reference_data: _Data,
+        comparison_data: _Data,
         block_size: int | None = None,
-        weights_x: ArrayLike | None = None,
-        weights_y: ArrayLike | None = None,
     ) -> Array:
         r"""
         Calculate maximum mean discrepancy.
 
-        If no weights are given for dataset ``y``, standard MMD is calculated. If
-        weights are given, weighted MMD is calculated. For both cases, if the size of
-        matrix blocks to process is less than the size of both datasets and if
-        ``weights_x`` is given in the weighted case, the calculation
-        is done block-wise to limit memory requirements.
+        If uniform weights are given for dataset ``reference_data``, standard MMD is
+        calculated. If non-uniform weights are given, weighted MMD is calculated. For
+        both cases, if the size of matrix blocks to process is less than the size of
+        both datasets and reference weights are not uniform in the weighted case, the
```

Comment:
I think this could be an issue in future - at present it's not possible to do the calculation block-wise with uniform weights, whereas before this change there was a difference between passing in _unweighted_ data (which could not be done block-wise) and _uniformly-weighted_ data (which could, as `weights_x is not None`).

Reply:
Could you write that ticket @tc85324 as you have more knowledge about the row_mean implementation?

Reply:
I think this issue is now resolved; we should correctly call `maximum_mean_discrepancy_block` if uniform weights are passed in along with a `block_size`. However, the docstring now needs to be updated to account for this.